### ISSUE TITLE
Follow-up lines duplicate characters and have weird indent

### ISSUE THREAD

**Issue by lars-vc** [zbirenbaum/copilot-cmp#66]
When accepting follow-up suggestions (on an empty line) weird behaviour occurs.
I have the following mapping setup in cmp:
```lua
    ["<Tab>"] = cmp.mapping(function(fallback)
        if cmp.visible() then
            cmp.confirm({ select = true, behavior = cmp.ConfirmBehavior.Replace })
        else
            fallback()
        end
    end, { "i", "c" }),
```

https://user-images.githubusercontent.com/57136208/235098170-06b51bc7-5578-4eb1-961b-6f180fef7352.mp4

**Comment by cuongn265** [comment]
can confirm same issue here

**Comment by zbirenbaum** [comment]
Please test #67 or checkout the `formatting-fixes` branch and see if it fixes your issue.

**Comment by lars-vc** [comment]
This indeed fixes the issues! I will leave this issue open til the pr is merged, so others can fix this too.

**Comment by codeDude64** [comment]
I confirm the branch `formatting-fixes` branch doesn't have this issue.

**Comment by benwoodward** [comment]
Fixed it for me! Thanks for your efforts on this plugin @zbirenbaum, wouldn't get half as much done without it.

**Comment by cuongn265** [comment]
can confirm it's fixed on the new branch, thank you so much @zbirenbaum

**Comment by pecavalheiro** [comment]
for me, this branch fixed the problem when inserting a new line, but made the line editing much worse. Now, when accepting the suggestion, the cursor moves to the previous line, breaking the overall indentation and syntax.

https://github.com/zbirenbaum/copilot-cmp/assets/5122010/147fe8c0-6bd2-410f-859f-2f337c9e7014

**Comment by zbirenbaum** [comment]
closed via #67
